### PR TITLE
States list API

### PR DESCRIPTION
**Pull request by prabhanshuguptagit**
**Story card:** [ch4631](https://app.clubhouse.io/simpledotorg/story/4631/add-an-api-to-list-available-states-in-a-deployment)

## Because

The app should be able to pick a state from the available states in a deployment.

## This addresses

Adds an API endpoint (`/api/v4/states`) that lists the available state names.

### CODE REVIEW

File: app/controllers/api/v4/states_controller.rb [hunk 1]
```diff
@@ -0,0 +1,12 @@
+class Api::V4::StatesController < APIController
+  include Api::V3::PublicApi
+
+  def sync_to_user
```
Comment (harimohanraj89) [inline app/controllers/api/v4/states_controller.rb:4]: This doesn't feel like a `sync` endpoint to me. Can we just make a simple `/states` endpoint for this?

File: app/controllers/api/v4/states_controller.rb [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+    state_names = Region.facility_regions
+      .joins("inner join regions states ON states.path @> regions.path and states.region_type = 'state'")
+      .distinct("states.name")
+      .pluck("states.name")
```
Comment (harimohanraj89) [inline app/controllers/api/v4/states_controller.rb:8]: What's the difference between this and `Region.state_regions`?

**Review by harimohanraj89 — CHANGES_REQUESTED** [review]
Thoughts below!

File: app/controllers/api/v4/states_controller.rb (hunk 1 above)
Comment (prabhanshuguptagit) [reply app/controllers/api/v4/states_controller.rb:4]: I went back and forth between a `sync_to_user` and `index`. Does it make sense to make this a sync endpoint (similar to the medications controller) for consistency's sake?

File: app/controllers/api/v4/states_controller.rb (hunk 2 above)
Comment (prabhanshuguptagit) [reply app/controllers/api/v4/states_controller.rb:8]: `Region.state_regions` can have states that don't have any facility under them. To get states that have at least one facility we could:
- Start from `Region.state_regions` and join a distinct facility on each state, filter states that have a facility
- Start from `Region.facility_regions` and get their states

The latter seemed simpler to write.

Comment (harimohanraj89) [reply app/controllers/api/v4/states_controller.rb:8]: Ah I see, the state regions are _every_ state in the country from the canonical list. Neat neat.

File: app/controllers/api/v4/states_controller.rb (hunk 1 above)
Comment (harimohanraj89) [reply app/controllers/api/v4/states_controller.rb:4]: Hmm, I'm kinda torn - I'd love to keep this uber simple. Wanna poke the team and see how we feel?

Comment (prabhanshuguptagit) [reply app/controllers/api/v4/states_controller.rb:4]: Makes sense 👍🏼 . I'm also leaning towards a regular GET endpoint.

File: swagger/v3/swagger.json [hunk 3]
```diff
@@ -2253,7 +2253,10 @@
             "voip_call",
             "manual_call",
             "sms",
-            "whatsapp"
+            "whatsapp",
+            "missed_visit_sms_reminder",
+            "missed_visit_whatsapp_reminder",
+            "imo"
```
Comment (harimohanraj89) [inline swagger/v3/swagger.json:2259]: Some older updates that never made it to docs yet, I guess?

**Review by harimohanraj89 — APPROVED** [review]
:shipit:

**Pull request merged**